Add Waterfall Docs (#36)
PaperMC/PaperDocs: source/waterfall/getting-started.rst

===============
Getting Started
===============

.. contents::
   :depth: 1
   :local:

> 

Reviewer: Section on what Waterfall and Travertine *are*. Note that Travertine is Waterfall but 1.7 support.

Reviewer 2: Specifically 1.7.10 support ^

Requirements
~~~~~~~~~~~~

> Waterfall and Travertine require **Java 8** or newer to run. It should run fine

Reviewer: ```suggestion
Waterfall and Travertine require **Java 8** or newer to run. It does run fine
```

on newer versions as well, such as Java 11, but certain plugins may have issues.

> Generally, LTS versions of Java are best supported, though you may have luck on newer versions.

Reviewer: ```suggestion
Generally, LTS versions of Java are targeted, though you may have luck on newer versions.
```

Migrating From BungeeCord
~~~~~~~~~~~~~~~~~~~~~~~~~

Waterfall and Travertine are drop in replacements for BungeeCord, you don't need to make any changes.

> Getting A Server Jar

Reviewer: ```suggestion
Getting A Proxy Jar
```

~~~~~~~~~~~~~~~~~~~~~

> Paper provides runnable server jars directly from our `website's downloads page <https://papermc.io/downloads#Waterfall>`_.

Reviewer: ```suggestion
Paper provides runnable proxy jars directly from our `website's downloads page <https://papermc.io/downloads#Waterfall>`_.
```

Click on the build number to download a file.

> Running The Server

Reviewer: ```suggestion
Running The Proxy
```

~~~~~~~~~~~~~~~~~~

> To run the server, simply start it up like any other Java application.

Reviewer: ```suggestion
To run the proxy, simply start it up like any other Java application.
```

Open your terminal, navigate to the saved location, and then run
``java -Xms512M -Xmx512M -jar waterfall-###.jar``

Aikar's recommended flags for Waterfall and Travertine are as follows:

> ``java -Xms512M -Xmx512M -XX:+UseG1GC -XX:G1HeapRegionSize=4M -XX:+UnlockExperimentalVMOptions``

Reviewer: I'd say we should rather link to Aikar's article in case they change. It also has info on Pterodactyl/Docker's memory limits and whatnot.

Reviewer 2: The flags aren't on the article (that I see), he put it in a pin in #waterfall-help I believe.

Reviewer: Bully Aikar into adding it then :>

Author: Yeah I did have the article linked originally but electronicboy said to remove them as they're just for Paper, we can either keep them until/if they're added or just remove them for the time being?

``-XX:+ParallelRefProcEnabled -XX:+AlwaysPreTouch -jar waterfall-###.jar``

> The amount of RAM can be set by changing the numbers in the ``-Xms`` and ``-Xmx``

Reviewer: ```suggestion
The amount of memory can be set by changing the numbers in the ``-Xms`` and ``-Xmx``
```

This is strictly speaking the heap memory for the JVM.

> arguments.

Reviewer: ```suggestion
flags.
```

> To configure your server, see the :doc:`../waterfall/configuration` page.

Reviewer: ```suggestion
To configure your proxy, see the :doc:`../waterfall/configuration` page.
```

> Updating The Server

Reviewer: ```suggestion
Updating The Proxy
```

~~~~~~~~~~~~~~~~~~~

> To update the server, first stop it safely by executing the ``end`` command

Reviewer: ```suggestion
To update the proxy, first stop it safely by executing the ``end`` command
```

> and then replace the old waterfall jar with a new one. That's it.

Reviewer: ```suggestion
and then replace the old proxy jar with a new one. That's it.
```

This is an article for two different proxies.
